lib/measured/unit.rb

```diff
@@ -64,9 +64,14 @@ def conversion_string
   end
 
   def parse_value(tokens)
-    tokens = tokens.split(" ", 2) if tokens.is_a?(String)
-
-    raise Measured::UnitError, "Cannot parse 'number unit' or [number, unit] formatted tokens from #{tokens}." unless tokens.size == 2
```

Comment:
This is all now much more robust. Error messaging is cleaner, and it allows units to be declared with more edge cases in mind.